File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int i=0;
        int j=heights.size()-1;
        int maxi=0;
        while(i<j)
        {
            maxi=max(maxi,(j-i)*min(heights[i],heights[j]));
            if(heights[i]<=heights[j])
            {
                i++;
            }
            else
            {
                j--;
            }
        }
        return maxi;
    }
};

File: Data Structures & Algorithms/max-area-of-island/submission-0.py
class Solution:
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        maxarea=float("-inf")
        def dfs(i,j):
            if i<0 or j<0 or i>len(grid)-1 or j>len(grid[i])-1:
                return 0
            if grid[i][j]!=1:
                return 0
            grid[i][j]=-1
            return 1+dfs(i+1,j)+dfs(i,j+1)+dfs(i-1,j)+dfs(i,j-1)
        for i in range(len(grid)):
            for j in range(len(grid[i])):
                maxarea=max(maxarea,dfs(i,j))
        return maxarea

        